logstash-core/src/main/java/org/logstash/plugins/discovery/PluginRegistry.java

```diff
@@ -95,24 +103,51 @@ private void discoverPlugins() {
                 }
             }
         }
+
+        // after loaded all plugins, check if aliases has to be provided
+        addAliasedPlugins(PluginType.FILTER, filters);
+        addAliasedPlugins(PluginType.OUTPUT, outputs);
+        addAliasedPlugins(PluginType.INPUT, inputs);
+        addAliasedPlugins(PluginType.CODEC, codecs);
     }
 
-    public Class<?> getPluginClass(PluginLookup.PluginType pluginType, String pluginName) {
-        if (pluginType == PluginLookup.PluginType.FILTER) {
-            return getFilterClass(pluginName);
+    private <T> void addAliasedPlugins(PluginType type, Map<String, Class<T>> pluginCache) {
+        final Map<String, Class<T>> aliasesToAdd = new HashMap<>();
+        for (Map.Entry<String, Class<T>> e : pluginCache.entrySet()) {
+            final String realPluginName = e.getKey();
+                final Optional<String> alias = aliasRegistry.aliasFromOriginal(type, realPluginName);
+                if (alias.isPresent()) {
+                    final String aliasName = alias.get();
+                    if (!pluginCache.containsKey(aliasName)) {
+                        // no real plugin with same alias name was found
+                        aliasesToAdd.put(aliasName, e.getValue());
```

Comment:
If we remove the log-at-lookup, we can place similar here to log when the alias is created:

```suggestion
                        aliasesToAdd.put(aliasName, e.getValue());
                        final String typeStr = pluginType.name().toLowerCase();
                        LOGGER.info("Plugin {}-{} is aliased as {}-{}", typeStr, realPluginName, typeStr, aliasName);
        }
```